app/actions/websocket/group.test.ts

```diff
@@ -0,0 +1,280 @@
+                        group_id: groupId,
+                        team_id: teamId,
+                    }),
+                },
+            } as WebSocketMessage;
+
+            jest.mocked(generateGroupAssociationId).mockReturnValue('association-id');
+
+            await handleGroupTeamDissociateEvent(serverUrl, msg);
+
+            const {database} = DatabaseManager.getServerDatabaseAndOperator(serverUrl);
+            expect(deleteGroupTeamById).toHaveBeenCalledWith(database, 'association-id');
+        });
+    });
+
+    describe('handleGroupChannelAssociatedEvent', () => {
+        it('should handle group channel association successfully', async () => {
+            const msg = {
+                data: {
+                    group_channel: JSON.stringify({
+                        group_id: groupId,
+                        channel_id: channelId,
+                    }),
+                },
+            } as WebSocketMessage;
+
+            await handleGroupChannelAssociatedEvent(serverUrl, msg);
+
+            const {operator} = DatabaseManager.getServerDatabaseAndOperator(serverUrl);
+            expect(operator.handleGroupChannelsForChannel).toHaveBeenCalledWith({
+                channelId,
+                groups: [{id: groupId}],
+                prepareRecordsOnly: false,
+            });
+        });
+    });
+
+    describe('handleGroupChannelDissociateEvent', () => {
+        it('should handle group channel dissociation successfully', async () => {
+            const msg = {
+                data: {
+                    group_channel: JSON.stringify({
+                        group_id: groupId,
+                        channel_id: channelId,
+                    }),
+                },
+            } as WebSocketMessage;
+
+            jest.mocked(generateGroupAssociationId).mockReturnValue('association-id');
+
+            await handleGroupChannelDissociateEvent(serverUrl, msg);
+
+            const {database} = DatabaseManager.getServerDatabaseAndOperator(serverUrl);
+            expect(deleteGroupChannelById).toHaveBeenCalledWith(database, 'association-id');
+        });
+    });
+
+    describe('error handling', () => {
+        it('should handle errors and fetch groups on failure', async () => {
+            const msg = {
+                broadcast: {
+                    team_id: teamId,
+                    channel_id: channelId,
+                    user_id: userId,
+                },
+                data: {
+                    group: 'invalid json',
+                },
+            } as WebSocketMessage;
+
+            await handleGroupReceivedEvent(serverUrl, msg);
+
+            expect(fetchGroupsForTeam).toHaveBeenCalledWith(serverUrl, teamId);
+            expect(fetchGroupsForChannel).toHaveBeenCalledWith(serverUrl, channelId);
+            expect(fetchGroupsForMember).toHaveBeenCalledWith(serverUrl, userId);
+        });
+
+        it('should handle invalid json on all events', async () => {
```

Comment:
what's the invalid json here? looks more like no valid data, but I'm not sure it is checking that and the expects looks like everything is in order

Reply:
The text 'invalid json' is the bad json data that triggers the throw/catch which is being tested here